shoop/front/middleware.py

```diff
@@ -85,6 +91,10 @@ def _set_timezone(self, request):
             timezone.activate(request.person.timezone)
             # TODO: Fallback to request.shop.timezone (and add such field)
 
+    def _get_maintenance_response(self, request):
+        if request.shop.maintenance_mode and not request.user.is_superuser:
+            return HttpResponse(loader.render_to_string("shoop/front/maintenance.jinja", request=request))
```

Comment:
Come to think of it, it might be a good idea to return a non-200 error code from here, to avoid, say, search engines indexing the maintenance page as regular content. 503 ("Service Unavailable") sounds like a good fit.

This will also simplify the test case; you can just look at the status code there.